extensions-contrib/opencensus-extensions/src/main/java/org/apache/druid/data/input/opencensus/protobuf/OpenCensusProtobufReader.java

```diff
@@ -80,13 +85,25 @@ public OpenCensusProtobufReader(
   @Override
   public CloseableIterator<InputRow> read()
   {
-    return CloseableIterators.withEmptyBaggage(readAsList().iterator());
+    Supplier<Iterator<InputRow>> supplier = Suppliers.memoize(() -> readAsList().iterator());
```

Comment:
why do we need to memoize this?

Reply:
Inherited change. Because we needed to initialize the iterator once I believe.